packages/monaco/src/browser/monaco-workspace.ts

```diff
@@ -175,14 +182,34 @@ export class MonacoWorkspace implements lang.Workspace {
     @inject(ProblemManager)
     protected readonly problems: ProblemManager;
 
+    protected _workspaceFolders: WorkspaceFolder[];
+    get workspaceFolders(): WorkspaceFolder[] {
+        return this._workspaceFolders;
+    }
+
     @postConstruct()
     protected init(): void {
         this.workspaceService.roots.then(roots => {
-            const rootStat = roots[0];
-            if (rootStat) {
-                this._rootUri = rootStat.uri;
+            this._workspaceFolders = roots.map((dir, index) => this.toWorkspaceFolder(dir.uri, index));
+            if (this._workspaceFolders.length !== 0) {
```

Comment:
Honestly i'm not sure that this predicate is correct, if open a Theia without a workspace, then `WorkspaceService` should be still ready. Wonder why we have it.